docs/spec/concepts.rst

```diff
@@ -0,0 +1,95 @@
+Type system concepts
```

Comment:
This feels like a grab-bag section, and it's oddly placed at the start of the document (why start with unions, when nothing else has been described yet)?

Reply:
This comes from Eric's outline in https://discuss.python.org/t/proposed-initial-typing-spec/39389/4, but I omitted the sections for which we don't have any text yet. I'm open to reorganizing this, but we can do that later too.

Reply:
Later is fine.